Dockerfile

```diff
@@ -1,31 +1,33 @@
-FROM neo4j:3.5.14-enterprise
+FROM neo4j:5.5.0-enterprise
 
-#ADD neo4j-graphql-3.5.0.1.jar plugins
-ADD apoc-3.5.0.1-all.jar plugins
+ADD apoc-5.5.0-core.jar plugins
+ADD apoc-5.5.0-extended.jar plugins
 
-#RUN echo 'dbms.unmanaged_extension_classes=org.neo4j.graphql=/graphql' >> /var/lib/neo4j/conf/neo4j.conf
-RUN echo 'dbms.security.auth_enabled=false' >> /var/lib/neo4j/conf/neo4j.conf
-RUN echo 'dbms.tx_state.memory_allocation=OFF_HEAP' >> /var/lib/neo4j/conf/neo4j.conf
-RUN echo 'dbms.tx_state.max_off_heap_memory=0' >> /var/lib/neo4j/conf/neo4j.conf
-RUN echo 'dbms.tx_log.rotation.retention_policy=false' >> /var/lib/neo4j/conf/neo4j.conf
-#RUN echo 'dbms.tx_log.preallocate=false' >> /var/lib/neo4j/conf/neo4j.conf
-RUN echo 'dbms.import.csv.buffer_size=134217728' >> /var/lib/neo4j/conf/neo4j.conf
-RUN echo 'dbms.connector.bolt.thread_pool_min_size=20' >> /var/lib/neo4j/conf/neo4j.conf
-RUN echo 'dbms.connector.bolt.thread_pool_max_size=100' >> /var/lib/neo4j/conf/neo4j.conf
-RUN echo 'dbms.connector.bolt.thread_pool_keep_alive=10m' >> /var/lib/neo4j/conf/neo4j.conf
-#RUN echo 'dbms.security.procedures.unrestricted=algo.*,apoc.*' >> /var/lib/neo4j/conf/neo4j.conf
-RUN echo 'cypher.min_replan_interval=120000ms' >> /var/lib/neo4j/conf/neo4j.conf
-RUN echo 'cypher.statistics_divergence_threshold=0.9' >> /var/lib/neo4j/conf/neo4j.conf
-#RUN cat /docker-entrypoint.sh | grep -v dbms.memory.pagecache.size > /temp_file && mv /temp_file /docker-entrypoint.sh && chmod +x /docker-entrypoint.sh
+RUN echo 'db.tx_state.memory_allocation=OFF_HEAP' >> /var/lib/neo4j/conf/neo4j.conf
+RUN echo 'server.memory.off_heap.max_size=0' >> /var/lib/neo4j/conf/neo4j.conf
+RUN echo 'db.import.csv.buffer_size=134217728' >> /var/lib/neo4j/conf/neo4j.conf
+RUN echo 'server.memory.heap.max_size=48GB' >> /var/lib/neo4j/conf/neo4j.conf
```

Comment:
Some of those are currently hard-coded in the main branch so I followed the same procedure.

I can remove them, we'd just need to be sure to set them properly elsewhere (devops repo? developers repo?) otherwise the database won't build on AWS. Since most people are building on AWS, I thought it was helpful to just set them here and folks can override them later if necessary.

Reply:
`neo4j-admin server memory-recommendation --memory=32g`
```
server.memory.heap.initial_size=12g
server.memory.heap.max_size=12g
server.memory.pagecache.size=12g
```
However once you cross the 32G mark
`neo4j-admin server memory-recommendation --memory=256g`
```
server.memory.heap.initial_size=31g
server.memory.heap.max_size=31g
server.memory.pagecache.size=205g
```
Java does 32bit memory addressing so once you cross the 32GB line you have to switch to 64bit addressing which litterly doubles the amount of memory you have to use to address the same space

Reply:
I would rather we leave the server to generate these an not us. You are right we will need to set them for local runs and maybe on the long running server but for loadering on the huge box lets let neo figure it out.

Reply:
We may though want to turn this on:
```
# It is also recommended turning out-of-memory errors into full crashes,
# instead of allowing a partially crashed database to continue running:
server.jvm.additional=-XX:+ExitOnOutOfMemoryError
```

Reply:
Dockerfile updated.